src/seedu/addressbook/Main.java

```diff
@@ -103,27 +115,58 @@ private void recordResult(CommandResult result) {
      * @param command user command
      * @return result of the command
      */
-    private CommandResult executeCommand(Command command)  {
+    private CommandResult executeCommand(Command command) {
         try {
+            checkIfFileExists(storage.getFile());
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
+
+        } catch (FileNotFoundException e) {
+            ui.showToUser(e.getMessage());
+            createNewFile();
```

Comment:
`createNewFile()` seems to be on a lower abstraction level then the rest of the other methods barring `checkIfFileExist(..)`.
Again `StorageFile` seems to be more qualified to handle creation of new file in this context.